modulemd/v2/include/modulemd-2.0/modulemd.h

```diff
@@ -172,9 +172,64 @@ G_BEGIN_DECLS
  * code-paths and support only the latest we understand. After that, it calls
  * validate() to ensure that the content that was read in was valid both
  * syntactically and referentially.
- */
-
-
+ * # Working with Libmodulemd API 2.0
+ * Steps to list the RPM API for a provided module NSVC in Python
+ * In the example below we assume we are dealing with metadata from only
+ * one repository ie fedora.
+ * We assume that the content of the YAML module metadata from the 
+ * repository is loaded into string variable "fedora_yaml"
+ *
+ * First step is to Import the libmodulemd 2.0 API into a Python script.
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * import gi
+ * gi.require_version('Modulemd', '2.0')
+ * from gi.repository import Modulemd
+ * ]|
+ *
+ * Second step is to Read repodata YAML from a file into an index.
+ * We load the metadata from the repository into ModulemdModuleIndex object
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * fedora_index = Modulemd.ModuleIndex.new()
+ * ret, failures = fedora_index.update_from_string(fedora_yaml)
+ * ]|
+ *
+ * Third step is to Query the Modulemd.ModuleIndex for the module with a 
+ * given name.
+ * If required the default stream for the perticulat module can also be 
+ * obtained by the second command.
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * module = fedora_index.get_module ('modulename')
+ * defaults = module.get_defaults()
+ * print ('Default stream for modulename is %s' % (
+ *        defaults.get_default_stream())
+ * ]|
+ *
+ * Fourth step is to Query the Modulemd.Module for the Modulemd.ModuleStream
+ * associated with the provided NSVC (name-stream-version-context identifier)
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * stream = module.get_stream_by_NSVC('modulestream', 1, 'deadbeef')
+ * ]|
+ *
+ * Last step is to Read the RPM API from the Modulemd.ModuleStream
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * api_list = stream.get_rpm_api()
```

Comment:
Describe that the `api_list` is a python list of strings containing package names.